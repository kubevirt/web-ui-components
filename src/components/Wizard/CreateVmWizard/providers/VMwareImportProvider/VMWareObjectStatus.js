import React from 'react';
import PropTypes from 'prop-types';

import { Alert, Spinner } from 'patternfly-react';

import {
  V2V_WMWARE_STATUS_CONNECTING,
  V2V_WMWARE_STATUS_CONNECTION_FAILED,
  V2V_WMWARE_STATUS_CONNECTION_SUCCESSFUL,
  V2V_WMWARE_STATUS_LOADING_VM_DETAIL,
  V2V_WMWARE_STATUS_LOADING_VM_DETAIL_FAILED,
  V2V_WMWARE_STATUS_LOADING_VMS_LIST,
  V2V_WMWARE_STATUS_LOADING_VMS_LIST_FAILED,
  V2V_WMWARE_STATUS_UNKNOWN,
  V2V_WMWARE_STATUS_CONNECTION_TO_VCENTER_FAILED,
} from '../../../../../utils/status/v2vVMware';
import VMwareStatusField from './VMwareStatusField';

const CheckingCredentials = () => (
  <VMwareStatusField>
    Checking vCenter Credentials...
    <Spinner loading size="sm" />
  </VMwareStatusField>
);

const LoadingData = () => (
  <VMwareStatusField>
    Connection successful. Loading data...
    <Spinner loading size="sm" />
  </VMwareStatusField>
);

const ConnectionFailed = () => (
  <VMwareStatusField>
    <Alert type="warning">Could not connect to vCenter using the provided credentials.</Alert>
  </VMwareStatusField>
);

const ConnectionFailedInfra = () => (
  <VMwareStatusField>
    <Alert type="warning">Can not verify vCenter credentials, connection to the V2V VMWare failed.</Alert>
  </VMwareStatusField>
);

const ConnectionSuccessful = () => <VMwareStatusField>Connection successful</VMwareStatusField>;

const ConnectionUnknown = () => <VMwareStatusField>Status unknown</VMwareStatusField>;

const ReadVmsListFailed = () => (
  <VMwareStatusField>
    <Alert type="warning">
      Connection succeeded but could not read list of virtual machines from the vCenter instance.
    </Alert>
  </VMwareStatusField>
);

const ReadVmDetailFailed = () => (
  <VMwareStatusField>
    <Alert type="warning">
      Connection succeeded but could not read details of the virtual machine from the vCenter instance.
    </Alert>
  </VMwareStatusField>
);

const vmwareStatusComponentResolver = {
  [V2V_WMWARE_STATUS_CONNECTING]: CheckingCredentials,
  [V2V_WMWARE_STATUS_CONNECTION_TO_VCENTER_FAILED]: ConnectionFailedInfra,
  [V2V_WMWARE_STATUS_CONNECTION_FAILED]: ConnectionFailed,
  [V2V_WMWARE_STATUS_LOADING_VMS_LIST]: LoadingData,
  [V2V_WMWARE_STATUS_LOADING_VMS_LIST_FAILED]: ReadVmsListFailed,
  [V2V_WMWARE_STATUS_LOADING_VM_DETAIL]: LoadingData,
  [V2V_WMWARE_STATUS_LOADING_VM_DETAIL_FAILED]: ReadVmDetailFailed,
  [V2V_WMWARE_STATUS_CONNECTION_SUCCESSFUL]: ConnectionSuccessful,
  [V2V_WMWARE_STATUS_UNKNOWN]: ConnectionUnknown,
};
// see onVmwareCheckConnection() for details
const VMWareObjectStatus = ({ status }) => {
  const StatusComponent = vmwareStatusComponentResolver[status] || ConnectionUnknown;

  return <StatusComponent />;
};

VMWareObjectStatus.defaultProps = {
  status: null,
};

VMWareObjectStatus.propTypes = {
  status: PropTypes.string,
};

export default VMWareObjectStatus;
