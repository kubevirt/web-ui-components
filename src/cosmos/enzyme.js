import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
