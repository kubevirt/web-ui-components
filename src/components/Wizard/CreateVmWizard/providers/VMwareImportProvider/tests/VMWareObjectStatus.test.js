import React from 'react';
import { mount } from 'enzyme';

import VMWareObjectStatus from '../VMWareObjectStatus';

const testVMWareProviderStatus = () => <VMWareObjectStatus />;

// TODO
describe('<VMWareObjectStatus />', () => {
  it('renders correctly', () => {
    const component = mount(testVMWareProviderStatus());
    expect(component).toMatchSnapshot();
  });
});
