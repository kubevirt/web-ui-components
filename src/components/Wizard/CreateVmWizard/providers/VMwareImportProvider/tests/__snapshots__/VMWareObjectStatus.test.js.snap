// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VMWareObjectStatus /> renders correctly 1`] = `
<VMWareObjectStatus
  status={null}
>
  <ConnectionUnknown>
    <VMwareStatusField
      id={null}
    >
      <div
        className="kubevirt-create-vm-wizard__import-vmware-connection-status"
        id={null}
      >
        Status unknown
      </div>
    </VMwareStatusField>
  </ConnectionUnknown>
</VMWareObjectStatus>
`;
